### PR TITLE
stagex refactor

### PR DESCRIPTION
**Pull request by lrvick**
## Highlights
- git lfs artifacts removed
- toolchain removed
- make flow dramatically simplified. Single file.
- all actual build steps now handled in overhauled containerfiles in src/images
- `make`: deterministically build all the things with stagex build container
- `make test`: run all tests in build container
- `make lint`: run automated linting from build container
  - disabled as current code does not pass linting
- `make format`: run rustfmt from build container
- `make docs`: run rustdocs from build container
- `make gen`: run "make -C proto" under stagex with stagex-provided tools
- CI refactored to use all of the above exactly as you would locally

### CODE REVIEW

File: .github/workflows/artifacts.yml [hunk 1]
```diff
@@ -178,8 +63,8 @@ jobs:
             --password "${{ secrets.GITHUB_TOKEN }}" \
             ghcr.io
           for image in ${images}; do
-            skopeo copy --all \
-              "oci-archive:./${image}.oci.x86_64.tar/${image}.oci.x86_64.tar" \
+            skopeo copy --all --dest-decompress \
+              "oci-archive:/home/runner/work/qos/qos/${image}.tar/${image}.tar" \
```
Comment (emostov) [inline .github/workflows/artifacts.yml]: interesting - how did you figure out the image was at this path?

Comment (lrvick) [reply .github/workflows/artifacts.yml]: It is the default path used in most examples, including ours elsewhere:
https://docs.github.com/en/actions/learn-github-actions/contexts

File: Makefile [hunk 2]
```diff
@@ -1,543 +1,131 @@
...
+			dest=out/build-base" \
+		--platform=linux/amd64 \
+		--tag qos-local/build-base \
+		$(NOCACHE_FLAG) \
+		-f src/images/Containerfile \
+		src/
+
+out/.build-base-loaded: out/build-base/index.json
+	cd out/build-base; \
+		tar -cf - . | docker load
```
Comment (james-callahan) [inline Makefile]: Doesn't this change the directory for the following `touch` command as well?

```suggestion
	(cd out/build-base; tar -cf - . | docker load)
```

or can you use the the `-C` argument to tar?
```
-C, --directory=DIR
              Change to DIR before performing any operations.  This option is order-sensitive, i.e. it affects all options that follow.
```

Comment (lrvick) [reply Makefile]: Macs dont have -C sadly. that is why I remove -C

Comment (james-callahan) [reply Makefile]: I know mac `env` doesn't have `-C` but I thought their `tar` would.

File: .github/workflows/artifacts.yml [hunk 3]
```diff
@@ -10,147 +10,30 @@ on:
...
-          make -d toolchain-restore-mtime toolchain out/${{ matrix.target }}
-
+          make out/${{ matrix.target }}
       - uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
         with:
           name: ${{ matrix.target }}
           path: out/${{ matrix.target }}
           retention-days: 1
 
-  upload_to_ecr:
```
Comment (emostov) [inline .github/workflows/artifacts.yml:106]: we need to keep this upload to ecr step. I can work on adding this back

**Pull request merged**